### DISCUSSION TITLE
[Question][Hw4] About sanitizer in HW4

### DISCUSSION THREAD

**Discussion by hsKim25** [prosyslab-classroom/cs348-information-security#563] [Q&A]
In HW4, before we do fuzzing, we should make executable file with 'instrumented.ll' which is comes from previous HW, sanitizer.
I think the work of fuzzing is depends on function calls of coverage and dbz_sanitizer, since fuzzing figure out crash or not by exit number, mainly div 0 in our HW.

I think, if the executable file is made by our HW3 sanitizer when HW4 is being graded, it is too dependent on our previous HW.
Although, in current simple c file and json_parser don't need dynamic checking, it could be exists some test cases that needs dynamic div0 checking in hidden test cases.

So, I wonder that making executable file is done by some master code (sample solution for HW3) or not. I hope the HW4 is not that dependent on previous homework.

Thank you.

**Comment by KihongHeo** [comment]
Evaluation will be done with your implementation in the HW4 repository.

If your implementation is not that strong, please try to improve it. TAs have already provided all test cases. If you need help, ask TAs or me in officer hours (not this public page please). Now we can answer to any questions regarding HW3. It will be good enough if your sanitizer passes all the provided test cases.

As in usual fuzzing tools, conceptually, HW3 is a part of HW4. I intentionally seperated them to give students intermediate feedback.

**Comment by hsKim25** [comment]
Thank you
